packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -0,0 +1,166 @@
+                    $"A new {modelProvider.Type:C} instance for mocking.",
+                    GetParameters(modelProvider));
+
+                var docs = new XmlDocProvider();
+                docs.Summary = modelProvider.XmlDocs?.Summary;
+                docs.Returns = new XmlDocReturnsStatement($"A new {modelProvider.Type:C} instance for mocking.");
+                foreach (var param in signature.Parameters)
+                {
+                    docs.Params.Add(new XmlDocParamStatement(param.Name, param.Description));
+                }
+
+                var statements = new MethodBodyStatements(
+                [
+                    .. GetCollectionInitialization(signature),
+                    MethodBodyStatement.EmptyLine,
+                    Return(New.Instance(modelProvider.Type, [.. GetCtorParams(signature)]))
+                ]);
+
+                methods.Add(new MethodProvider(signature, statements, this, docs));
+            }
+            return [.. methods];
+        }
+
+        private IReadOnlyList<ValueExpression> GetCtorParams(MethodSignature signature)
+        {
+            var expressions = new List<ValueExpression>(signature.Parameters.Count);
+            foreach (var param in signature.Parameters)
+            {
+                if (param.Type.IsList)
+                {
+                    expressions.Add(param.NullConditional().ToList());
+                }
+                else
+                {
+                    expressions.Add(param);
+                }
+            }
+            expressions.Add(Null);
+            return [.. expressions];
+        }
+
+        private IReadOnlyList<MethodBodyStatement> GetCollectionInitialization(MethodSignature signature)
+        {
+            var statements = new List<MethodBodyStatement>();
+            foreach (var param in signature.Parameters)
+            {
+                if (param.Type.IsList)
+                {
+                    statements.Add(param.Assign(New.Instance(new CSharpType(typeof(List<>), param.Type.Arguments))).Terminate());
+                }
+                else if (param.Type.IsDictionary)
+                {
+                    statements.Add(param.Assign(New.Instance(new CSharpType(typeof(Dictionary<,>), param.Type.Arguments))).Terminate());
+                }
+            }
+            return [.. statements];
+        }
+
+        private IReadOnlyList<ParameterProvider> GetParameters(TypeProvider modelProvider)
+        {
+            var parameters = new List<ParameterProvider>(modelProvider.Properties.Count);
+            foreach (var property in modelProvider.Properties)
+            {
+                if (property.Modifiers.HasFlag(MethodSignatureModifiers.Internal))
+                    continue;
+
+                parameters.Add(GetModelFactoryParam(property.AsParameter));
+            }
+            return [.. parameters];
+        }
+
+        private ParameterProvider GetModelFactoryParam(ParameterProvider parameter)
```

Comment:
could we just simply have `property.AsParameter.GetInputXXX()` (I forgot the name of this API but parameter has an API to returns its input variant)

Reply:
this is more than a type change it also sets the default expression to `Default`